### PR TITLE
Add custom http client option to GraphQLClient

### PR DESCRIPTION
**Pull request by jainpiyush19**
Currently there is no way to customize the underlying HTTPClient used by the GraphQLClient.

There are many different usecases where HTTPClient needs to be customzied like introducing circuit breakers or custom roundtrippers.

I would like to introduce a new option to set custom HTTPClient.

### CODE REVIEW

**Comment by jainpiyush19** [comment]
https://github.com/movio/bramble/issues/121

**Review by lucianjon — CHANGES_REQUESTED** [review]
Thanks @jainpiyush19! Just a couple of comments around the test but otherwise LGTM.

File: config.go [hunk 1]
```diff
@@ -264,7 +267,11 @@ func (c *Config) Init() error {
 		services = append(services, NewService(s))
 	}
 
-	queryClient := NewClient(WithMaxResponseSize(c.MaxServiceResponseSize), WithUserAgent(GenerateUserAgent("query")))
+	queryClientOptions := []ClientOpt{WithMaxResponseSize(c.MaxServiceResponseSize), WithUserAgent(GenerateUserAgent("query"))}
+	if c.QueryHTTPClient != nil {
```
[the remark this answers is not in the document]
Comment (pkqk) [reply config.go:271]: Making a custom `main.go` would probably also be enough for now but a custom plugin is the most flexible option.

**Review by pkqk — APPROVED** [review]

**Comment by pkqk** [comment]
Thanks @jainpiyush19, we'll merge this in as it is, but an example plugin that shows the custom HTTPClient being used would also be appreciated.

**Pull request merged**